### ISSUE TITLE
Question about adding a password for each attachment

### ISSUE THREAD

**Issue by costescuandrei** [tdulcet/Thunderbird-Send#8]
I see that the service itself does support passwords for each file attachment/upload.
That would be a welcome feature on the plugin as well - as the wiki also states.

I created this issue in order to find out:

1. if this is still the repo where development happens for the plugin and future versions will end up in Thunderbird addons section. Just in case I am in the mood to learn Thunderbird plugin development a bit and make a pull req...
2. if I want to donate to 'stimulate' :) this feature, where should the donation go to? The info and donate pages direct towards @tdulcet but a few other places say that @timvisee is maintaining it (although I don't see a fork from him). So where do donations for future development go? Should the info/donate pages be updated?

**Comment by costescuandrei** [comment]
Ah @timvisee maintains the server part and this repo of @tdulcet is the official one for the Thunderbird plugin. Did I get it right this time?

**Comment by tdulcet** [comment]
> That would be a welcome feature on the plugin as well - as the wiki also states.

Yes, my add-on does not currently support this feature, but it is something I am considering for a future version.

> 1. if this is still the repo where development happens for the plugin and future versions will end up in Thunderbird addons section. Just in case I am in the mood to learn Thunderbird plugin development a bit and make a pull req...

Yes, this is the correct repository for the "Filelink provider for Send" Thunderbird add-on. Pull requests are always welcome! 😉 The Send server code is maintained by @timvisee here: https://github.com/timvisee/send, which I would recommend looking at if you are interested in adding this password feature to the add-on.

> 2. if I want to donate to 'stimulate' :) this feature, where should the donation go to? The info and donate pages direct towards @tdulcet but a few other places say that @timvisee is maintaining it (although I don't see a fork from him). So where do donations for future development go?

I maintain this Thunderbird add-on, while @timvisee maintains the Send server code and hosts the public https://send.vis.ee Send instance. I am always grateful for any donations that allow me to continue development of the add-on.

> Should the info/donate pages be updated?

Yeah, it seems I should update the README and ATN listing to make this clearer. Suggestions are welcome...

> Ah @timvisee maintains the server part and this repo of @tdulcet is the official one for the Thunderbird plugin. Did I get it right this time?

Yes, that is correct! Please let me know if I answered all of your questions.

**Comment by costescuandrei** [comment]
Don't feel any obligations for a small donation that may come :).
Take it as a thank you for the current plugin.

If I start working on it I will say that here, but the chance is small in the near future.

**Comment by timvisee** [comment]
Send (the server) does support passwords.

@tdulcet wasn't the problem that a password box is hard to implement through the File Link API?

**Comment by tdulcet** [comment]
Thank you @costescuandrei for your generous donation! 🙏

> wasn't the problem that a password box is hard to implement through the File Link API?

I was hoping that Thunderbird would implement something like what I requested in [bug 1711446](https://bugzilla.mozilla.org/show_bug.cgi?id=1711446) or [bug 1752114](https://bugzilla.mozilla.org/show_bug.cgi?id=1752114), which would have provided a better UI/UX for this. However, the add-on should still be able to ask for an optional password in the existing popup window that it opens.

The main issue was that I did not know if there was any interest this feature and it would obviously require some significant changes to the add-on. Based on this and #6, it seems there is some interest, so I will definitely consider adding it in the next version...

**Comment by tdulcet** [comment]
I looked into this password feature some more and it is actually not as difficult as I had originally thought, so I was able to implement a quick prototype:
![image](https://user-images.githubusercontent.com/15305150/218269232-d74fdd00-f926-4368-9fec-647f7deffaea.png)

Assuming testing goes well, this will be included with the next version, likely in time for the upcoming Thunderbird 115 "Supernova" release. Feedback is welcome. I might also add a button to automatically generate a strong password, since Thunderbird of course does not have that functionality as most browsers do.
